rwengine/src/engine/Garage.hpp

```diff
@@ -0,0 +1,111 @@
+#ifndef _RWENGINE_Garage_HPP_
+#define _RWENGINE_Garage_HPP_
+
+#include <glm/glm.hpp>
+#include <glm/gtc/quaternion.hpp>
+#include <rw/defines.hpp>
+
+class PlayerController;
+
+class GameWorld;
+
+class CharacterObject;
+class InstanceObject;
+class GameObject;
+
+class Garage {
+private:
+    InstanceObject* doorObject = nullptr;
+    InstanceObject* secondDoorObject = nullptr;
+
+    bool active = true;
+
+    float garageTimer = 0.f;
+
+    bool swingType = false;
+
+    glm::vec3 startPosition;
+    glm::vec3 startPositionSecondDoor;
+
+    bool needsToUpdate = false;
+
+    float fraction = 0.f;
+    float step = 3.f;  // this should be adjusted somehow
+                       // to look similar to original game
+    float doorHeight = 4.f;
+
+    float getDistanceToGarage(glm::vec3 point);
+
+    bool shouldOpen();
+    bool shouldClose();
+    bool shouldStopOpening();
+    bool shouldStopClosing();
+
+    void doOnOpenEvent();
+    void doOnCloseEvent();
+    void doOnStartOpeningEvent();
+    void doOnStartClosingEvent();
+
+    void updateDoor();
+
+public:
+    enum class Type {
```

Comment:
nit: if these are constants the real game uses there could be a comment about this here.